rest-api/src/test/java/com/cloudbees/workflow/rest/endpoints/JobAndRunAPITest.java

```diff
@@ -52,14 +56,67 @@
 import java.util.List;
 
 /**
- * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ * Test the raw job/run APIs
  */
 public class JobAndRunAPITest {
 
     @Rule
     public JenkinsRule jenkinsRule = new JenkinsRule();
 
     @Test
+    public void testBlockStage() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "Blocky job");
+
+        job.setDefinition(new CpsFlowDefinition("" +
+                "node {" +
+                "   stage ('Build') { " +
+                "     echo ('Building'); " +
+                "   } \n" +
+                "   stage ('Test') { " +
+                "     echo ('Testing'); " +
+                "   } \n" +
+                "   stage ('Deploy') { " +
+                "     writeFile file: 'file.txt', text:'content'; " +
+                "     archive(includes: 'file.txt'); " +
+                "     echo ('Deploying'); " +
+                "   } \n" +
+                "}"));
+
+        QueueTaskFuture<WorkflowRun> build = job.scheduleBuild2(0);
+        jenkinsRule.assertBuildStatusSuccess(build);
+
+        JenkinsRule.WebClient webClient = jenkinsRule.createWebClient();
+
+        assertBasicJobInfoOkay(job, webClient);
+        assertArtifactsEndpointOkay(job, webClient);
+        assertChangesetsEndpointOkay(job, webClient);
+
+        /** Verify the runs show right stages as per {@link #assertBasicRunInfoOkay(WorkflowJob, JenkinsRule.WebClient)} */
```

Comment:
It won't generate true Javadocs, but it points someone helpfully at the right place (and IIRC IntelliJ is sometimes smart enough to do links inside comments anyway).